File: labs/ooplab7/lab7sharp/LinkedList.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;

namespace lab7
{
    public struct LinkedList : IEnumerable
    {
        Node head;
        Node tail;
        int count; 

        public void Removenegative()
        {
            int counter = 1;
            foreach (double item in this)
            {
                if (item < 0)
                {
                    Remove(item);
                }
                counter++;
            }
        }
        public void AppendFirst(double data)
        {
            Node node = new Node(data);
            node.Next = head;
            head = node;
            if (count == 0)
                tail = head;
            count++;
        }
        public bool Remove(double data)
        {
            Node current = head;
            Node previous = null;

            while (current != null)
            {
                if (current.Data.Equals(data))
                {
                    if (previous != null)
                    {
                        previous.Next = current.Next;
                        if (current.Next == null)
                            tail = previous;
                    }
                    else
                    {
                        head = head.Next;
                        if (head == null)
                            tail = null;
                    }
                    count--;
                    return true;
                }

                previous = current;
                current = current.Next;
            }
            return false;
        }
        public int NumberOfElementsBiggerThanAvg()
        {
            double avg = 0;
            foreach (double item in this)
            {
                avg = avg + item;
            }
            avg /= Count;

            int counter = 0;

            foreach (double item in this)
            {
                if (item > avg)
                {
                    counter++;
                }
            }
            return counter;
        }

        public int Count { get { return count; } }
        public bool IsEmpty { get { return count == 0; } }
        IEnumerator IEnumerable.GetEnumerator()
        {
            Node current = head;
            while (current != null)
            {
                yield return current.Data;
                current = current.Next;
            }
        }
    }
    public class Node
    {
        public Node(double data)
        {
            Data = data;
        }
        public double Data { get; set; }
        public Node Next { get; set; }
    }
}

File: ooplab7/lab7_c++/LinkedList.cpp
#include "LinkedList.h"

double LinkedList::MedianValue() {
	double sum = 0.0;
	int count = 0;
	Node* temp = head;
	while (temp != NULL) {
		sum += temp->Data;
		count++;
		temp = temp->Link;
	}
	return sum / count;
}
void LinkedList::DeleteEvenElements() {
	Node* next = head->Link->Link;
	Node* after = head->Link->Link->Link;
	head->Link = head->Link->Link;
	while (next != NULL && after != NULL) {
		cout << next->Data;
		if (next->Data < 0) {
			cout << next->Data;
			next->Link = after->Link;
			after = NULL;
			next = next->Link;
			if (next != NULL) {
				after = next->Link;
			}
		}
	}
	//while (temp != NULL) {
	//	if (temp->Data < 0) {
	//		temp = NULL;
	//	}
	//	temp = temp->Link;
	//}
}
void LinkedList::PrintList() {
	Node* temp = head;
	while (temp != NULL) {
		cout << temp->Data << "\n";
		temp = temp->Link;
	}
}

int LinkedList::FindNumberOfElementsGreaterThanMedian() {
	int count = 0;
	double median = MedianValue();
	Node* temp = head;
	while (temp != NULL) {
		if (temp->Data > median) {
			count++;
		}
		temp = temp->Link;
	}
	return count;
}